SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/proof/ProofRules.java

```diff
@@ -0,0 +1,1133 @@
+ *
+ * SMTInterpol is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with SMTInterpol.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.uni_freiburg.informatik.ultimate.smtinterpol.proof;
+
+import java.math.BigInteger;
+import java.util.BitSet;
+import java.util.HashSet;
+
+import de.uni_freiburg.informatik.ultimate.logic.AnnotatedTerm;
+import de.uni_freiburg.informatik.ultimate.logic.Annotation;
+import de.uni_freiburg.informatik.ultimate.logic.ApplicationTerm;
+import de.uni_freiburg.informatik.ultimate.logic.ConstantTerm;
+import de.uni_freiburg.informatik.ultimate.logic.FunctionSymbol;
+import de.uni_freiburg.informatik.ultimate.logic.LambdaTerm;
+import de.uni_freiburg.informatik.ultimate.logic.PrintTerm;
+import de.uni_freiburg.informatik.ultimate.logic.QuantifiedFormula;
+import de.uni_freiburg.informatik.ultimate.logic.Rational;
+import de.uni_freiburg.informatik.ultimate.logic.SMTLIBConstants;
+import de.uni_freiburg.informatik.ultimate.logic.Sort;
+import de.uni_freiburg.informatik.ultimate.logic.Term;
+import de.uni_freiburg.informatik.ultimate.logic.TermVariable;
+import de.uni_freiburg.informatik.ultimate.logic.Theory;
+
+public class ProofRules {
+	// the function symbols
+	public final static String RES = "res";
+	public final static String ASSUME = "assume";
+	public final static String AXIOM = "axiom";
+
+	// the axioms
+	public final static String ORACLE = "oracle";
+	public final static String FALSEE = "false-";
+	public final static String TRUEI = "true+";
+	public final static String NOTI = "not+";
+	public final static String NOTE = "not-";
+	public final static String ORI = "or+";
+	public final static String ORE = "or-";
+	public final static String ANDI = "and+";
+	public final static String ANDE = "and-";
+	public final static String IMPI = "=>+";
+	public final static String IMPE = "=>-";
+	public final static String IFFI1 = "=+1";
+	public final static String IFFI2 = "=+2";
+	public final static String IFFE1 = "=-1";
+	public final static String IFFE2 = "=-2";
+	public final static String XORI = "xor+";
+	public final static String XORE = "xor-";
+	public final static String FORALLI = "forall+";
+	public final static String FORALLE = "forall-";
+	public final static String EXISTSI = "exists+";
+	public final static String EXISTSE = "exists-";
+	// equality chains of length >=3
+	public final static String EQI = "=+";
+	public final static String EQE = "=-";
+	public final static String DISTINCTI = "distinct+";
+	public final static String DISTINCTE = "distinct-";
+
+	public final static String ITE1 = "ite1";
+	public final static String ITE2 = "ite2";
+	public final static String REFL = "refl";
```

Comment:
The text documents suggest that refl is not part of the new proof rules and should be replaced with trans.